src/mime.tsx

```diff
@@ -0,0 +1,184 @@
+import { IRenderMime } from '@jupyterlab/rendermime-interfaces';
+// import { mystIcon } from './icon';
+import { markdownParse } from './myst';
+import {
+  DOITransformer,
+  enumerateTargetsPlugin,
+  footnotesPlugin,
+  getFrontmatter,
+  GithubTransformer,
+  keysPlugin,
+  linksPlugin,
+  mathPlugin,
+  ReferenceState,
+  resolveReferencesPlugin,
+  RRIDTransformer,
+  WikiTransformer
+} from 'myst-transforms';
+import { VFile } from 'vfile';
+import { PageFrontmatter, validatePageFrontmatter } from 'myst-frontmatter';
+import { internalLinksPlugin, linkFactory } from './links';
+import { FrontmatterBlock } from '@myst-theme/frontmatter';
+import {
+  ReferencesProvider,
+  TabStateProvider,
+  Theme,
+  ThemeProvider
+} from '@myst-theme/providers';
+import { useParse } from 'myst-to-react';
+import { renderers } from './renderers';
+import { References } from 'myst-common';
+import { ReactWidget, UseSignal } from '@jupyterlab/apputils';
+import { unified } from 'unified';
+import { Signal } from '@lumino/signaling';
+import React from 'react';
+
+/**
+ * The MIME type for Markdown.
+ */
+export const MIME_TYPE = 'text/markdown';
+
+type MySTState = {
+  references: References;
+  frontmatter: PageFrontmatter;
+  mdast: any;
+};
+
+/**
+ * A mime renderer for displaying Markdown with embedded latex.
+ */
+export class RenderedMySTMarkdown
+  extends ReactWidget
+  implements IRenderMime.IRenderer
+{
+  /**
+   * Construct a new rendered markdown widget.
+   *
+   * @param options - The options for initializing the widget.
+   */
+  constructor(options: IRenderMime.IRendererOptions) {
+    super();
+    this._state = null;
+    this.resolver = options.resolver;
+    this.linkHandler = options.linkHandler;
+    this.node.dataset['mimeType'] = MIME_TYPE;
+    this.addClass('myst-RenderedMySTMarkdown');
+    console.log('Rendered markdown');
+  }
+
+  /**
+   * The resolver object.
+   */
+  readonly resolver: IRenderMime.IResolver | null;
+
+  private _state: MySTState | null;
+  private _stateChanged = new Signal<this, void>(this);
+
+  /**
+   * The link handler.
+   */
+  readonly linkHandler: IRenderMime.ILinkHandler | null;
+
+  render(): JSX.Element {
+    return (
+      <UseSignal signal={this._stateChanged} initialSender={this}>
+        {(): JSX.Element => {
+          if (this._state === null) {
+            return <div>Missing</div>;
+          }
+          const { references, frontmatter, mdast } = this._state;
+          const children = useParse(mdast as any, renderers);
+
+          return (
+            <ThemeProvider
+              theme={Theme.light}
+              Link={linkFactory(this.resolver, this.linkHandler)}
+              renderers={renderers}
+            >
+              <TabStateProvider>
+                <ReferencesProvider
+                  references={references}
```

Comment:
Is this intended to share the state with other cells, e.g. can you reference an equation in the notebook with this?

It that is the case, then this needs to be the shared references, which gets a bit tricky (maybe?)

Reply:
Yes it does! When I wrote this PR initially, I just wanted to get the majority of the boilerplate on the page. The semantics are probably in need of some work :)